cg/apps/demultiplex/sample_sheet/sample_sheet_validator.py

```diff
@@ -0,0 +1,172 @@
+"""Module with validator classes for the sample sheet."""
+
+import logging
+from pathlib import Path
+from typing import Type
+
+from cg.apps.demultiplex.sample_sheet.override_cycles_validator import OverrideCyclesValidator
+from cg.apps.demultiplex.sample_sheet.read_sample_sheet import (
+    get_flow_cell_samples_from_content,
+    get_raw_samples,
+    get_sample_type,
+    validate_samples_unique_per_lane,
+)
+from cg.apps.demultiplex.sample_sheet.sample_models import FlowCellSample, FlowCellSampleBCLConvert
+from cg.apps.demultiplex.sample_sheet.sample_sheet_models import SampleSheet
+from cg.constants.constants import FileFormat
+from cg.constants.demultiplexing import NAME_TO_INDEX_SETTINGS, SampleSheetBCLConvertSections
+from cg.exc import OverrideCyclesError, SampleSheetError
+from cg.io.controller import ReadFile
+
+LOG = logging.getLogger(__name__)
+
+
+class SampleSheetValidator:
+    """Class for validating the content of a sample sheet."""
```

Comment:
Design issue: separation of concerns. It seems like this module sets a lot of data? I would expect it to have a single method called `validate` which accepts a single argument like a path to a sample sheet (or a parsed sample sheet)

Reply:
#2963